From ce453596895fd39a12af1a90014fc1ef0301f9b3 Mon Sep 17 00:00:00 2001
From: ctang <ctang.ma@gmail.com>
Date: Thu, 16 Jul 2015 14:34:28 -0400
Subject: [PATCH 0199/1164] CDH-29558: HIVE-11216: UDF GenericUDFMapKeys
 throws NPE when a null map value is passed in
 (Yibing Shi via Chaoyu Tang, reviewed by Szehon
 Ho

---
 .../hive/ql/udf/generic/GenericUDFMapKeys.java     |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFMapKeys.java b/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFMapKeys.java
index 1d4bcef..b3c8962 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFMapKeys.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFMapKeys.java
@@ -19,6 +19,7 @@
 package org.apache.hadoop.hive.ql.udf.generic;
 
 import java.util.ArrayList;
+import java.util.Map;
 
 import org.apache.hadoop.hive.ql.exec.Description;
 import org.apache.hadoop.hive.ql.exec.UDFArgumentException;
@@ -61,7 +62,10 @@ public ObjectInspector initialize(ObjectInspector[] arguments)
   public Object evaluate(DeferredObject[] arguments) throws HiveException {
     retArray.clear();
     Object mapObj = arguments[0].get();
-    retArray.addAll(mapOI.getMap(mapObj).keySet());
+    Map<?,?> mapVal = mapOI.getMap(mapObj);
+    if (mapVal != null) {
+      retArray.addAll(mapVal.keySet());
+    }
     return retArray;
   }
 
-- 
1.7.9.5

